src/Microsoft.ML.Transforms/WhiteningTransform.cs

```diff
@@ -185,15 +215,16 @@ public void Save(ModelSaveContext ctx)
 
         private const Mkl.Layout Layout = Mkl.Layout.RowMajor;
 
-        // Stores whitening matrix as Float[] for each column.
-        private readonly Float[][] _models;
-        // Stores inverse ("recover") matrix as Float[] for each column. Temporarily internal as it's used in unit test.
+        // Stores whitening matrix as float[] for each column. _models[i] is the whitening matrix of the i-th input column.
+        private readonly float[][] _models;
```

Comment:
>_models [](start = 35, length = 7)

This doesn't have to be part of this PR, so feel free to create new issue.
We need to have way which allow user to access internal models of whitening transform and let user apply this internal models to suitable arrays.
Basically we need to have class which would expose FillValues and have inside itself _models values.
Plus we need to have onFit function for pigsty to pass this class out.

Reply:
I made a new issue on that.

---
In reply to: [231675981](https://github.com/dotnet/machinelearning/pull/1452#discussion_r231675981) [](ancestors = 231675981)